backend/src/apiserver/template/v2_template.go

```diff
@@ -77,9 +77,20 @@ func (t *V2Spec) ScheduledWorkflow(modelJob *model.Job) (*scheduledworkflow.Sche
 		}
 	}
 
+	var pipeline_options argocompiler.Options
+	for _, platform := range t.platformSpec.Platforms {
+		if platform.PipelineConfig.SemaphoreKey != "" || platform.PipelineConfig.MutexName != "" {
```

Comment:
1. We should specifically check for the "kubernetes" platform here. There could be others, and we would ignore all others
2. Check that if you put some other non-kubernetes platform in there that this line doesn't cause a panic. I don't think it will, but just check :)

Reply:
Done. It now iterates over the `platformSpec.Platforms` map and adds a check to ensure that only the "kubernetes" platform is processed. It also includes the nil checks for platform and `platform.PipelineConfig` to prevent nil pointer dereference errors.